plugins/wallabag.koplugin/main.lua

```diff
@@ -315,13 +334,86 @@ function Wallabag:getBearerToken()
     end
 end
 
+--- Get a JSON formatted list of articles from the server.
+-- The list should have self.article_per_sync item, or less if an error occured.
+-- If filter_tag is set, only articles containing this tag are queried.
+-- If ignore_tags is defined, articles containing either of the tags are skipped.
 function Wallabag:getArticleList()
     local filtering = ""
     if self.filter_tag ~= "" then
         filtering = "&tags=" .. self.filter_tag
     end
-    local articles_url = "/api/entries.json?archive=0&perPage=" .. self.articles_per_sync .. filtering
-    return self:callAPI("GET", articles_url, nil, "", "")
+
+    local article_list = {}
+    local page = 1
+    -- query the server for articles until we hit our target number
+    while #article_list < self.articles_per_sync do
+        -- get the JSON containing the article list
+        local articles_url = "/api/entries.json?archive=0"
+                          .. "&page=" .. page
+                          .. "&perPage=" .. self.articles_per_sync
+                          .. filtering
+        local articles_json = self:callAPI("GET", articles_url, nil, "", "")
+
+        if not articles_json then
+            -- we may have hit the last page, there are no more articles
+            logger.dbg("Wallabag: couldn't get page #", page)
+            break -- exit while loop
+        end
+
+        -- We're only interested in the actual articles in the JSON
+        -- build an array of those so it's easier to manipulate later
+        local new_article_list = {}
+        for _, article in ipairs(articles_json._embedded.items) do
+            table.insert(new_article_list, article)
+        end
+
+        -- Apply the filters
+        new_article_list = self:filterIgnoredTags(new_article_list)
+
+        -- Append the filtered list to the final article list
+        local num_articles_left = self.articles_per_sync - #article_list
+        local num_articles_to_append = math.min(#new_article_list, num_articles_left)
+        for i=0, num_articles_to_append, 1 do
```

Comment:
```suggestion
        for i=1, num_articles_to_append, 1 do
```
This isn't very Lua. I know, I hate it when people say some particular construct isn't "Pythonic" but having some random C-style 0 in there is just confusing both to readers and the built-in table-handling functions.

Reply:
Also I specifically hate it when people say that without specifying what *is* and I think I dodged that bullet. ;-)

Reply:
Damn that's right, Lua indexes start at 1!
I used `ipairs` instead.